disvae/models/losses.py

```diff
@@ -537,6 +616,15 @@ def _get_log_pz_qz_prodzi_qzCx(latent_sample, latent_dist, n_data, is_mss=True):
         mat_log_qz = mat_log_qz + log_iw_mat.view(batch_size, batch_size, 1)
 
     log_qz = torch.logsumexp(mat_log_qz.sum(2), dim=1, keepdim=False)
-    log_prod_qzi = torch.logsumexp(mat_log_qz, dim=1, keepdim=False).sum(1)
+    log_prod_qzi = torch.logsumexp(mat_log_qz, dim=1, keepdim=False).sum(dim=1)
+
+    knc = _sum_along_dim_exclude_self(mat_log_qz, dim=2)
+    log_qz_knc = torch.logsumexp(knc, dim=1, keepdim=False).sum(dim=1)
+
+    return log_pz, log_qz, log_prod_qzi, log_q_zCx, log_qz_knc
 
-    return log_pz, log_qz, log_prod_qzi, log_q_zCx
+def _sum_along_dim_exclude_self(tensor_in, dim):
```

Comment:
nice implementation, should be documented, remove the leading underscore and moved to `math.py` that's the kind of function we might use again not only in losses